### PR TITLE
End-to-end tests for Tac skills implemented. AEA-359

### PR DESCRIPTION
**Pull request by panasevychol**
## Proposed changes

End-to-end tests for missing skills packages added
#685 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_packages/test_skills/test_tac.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+class TestTacSkills:
+    """Test that tac skills work."""
+
+    @pytest.fixture(autouse=True)
+    def _start_oef_node(self, network_node):
+        """Start an oef node."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the test class."""
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills/test_tac.py]: what about the tac controller?

File: tests/test_packages/test_skills/test_tac.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+        result = self.runner.invoke(
+            cli, [*CLI_LOG_OPTION, "create", self.agent_name_one], standalone_mode=False
+        )
+        assert result.exit_code == 0
+        result = self.runner.invoke(
+            cli, [*CLI_LOG_OPTION, "create", self.agent_name_two], standalone_mode=False
+        )
+        assert result.exit_code == 0
+
+        # add packages for agent one and run it
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills/test_tac.py]: remove `and run it`

File: tests/test_packages/test_skills/test_tac.py [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        # install
+        result = self.runner.invoke(
+            cli, [*CLI_LOG_OPTION, "install"], standalone_mode=False
+        )
+        assert result.exit_code == 0
+
+        os.chdir(self.t)
+
+
+        # add packages for agent two and run it
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills/test_tac.py]: remove `and run it`

File: tests/test_packages/test_skills/test_tac.py [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+        # add tac_negotiation skill
+        result = self.runner.invoke(
+            cli,
+            [*CLI_LOG_OPTION, "add", "skill", "fetchai/tac_negotiation:0.1.0"],
+            standalone_mode=False,
+        )
+        assert result.exit_code == 0
+
+        # Load the tac_participation yaml file and manually insert the things we need
+        yaml_path = os.path.join(
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills/test_tac.py]: there would also be config set commands from aea cli to set these attributes.

File: tests/test_packages/tools_for_testing.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    def teardown_class(cls):
+        """Teardowm the test."""
+        os.chdir(cls.cwd)
+        try:
+            shutil.rmtree(cls.t)
+        except (OSError, IOError):
+            pass
+
+    @staticmethod
+    def disable_ledger_tx(vendor_name, item_type, item_name):
```
Comment (DavidMinarsch) [inline tests/test_packages/tools_for_testing.py]: This is in my opinion misplaced here as it is not generic. Also, as mentioned in a previous comment, should use `aea config set ...` to update configs

Comment (DavidMinarsch) [reply tests/test_packages/tools_for_testing.py]: So what you could do is create a wrapper to call the config set command.

Comment (panasevychol) [reply tests/test_packages/tools_for_testing.py]: Done.

**Pull request merged**